WordPress/src/main/java/org/wordpress/android/ui/prefs/EmptyViewRecyclerView.java

```diff
@@ -0,0 +1,71 @@
+package org.wordpress.android.ui.prefs;
+
+import android.content.Context;
+import android.support.v7.widget.RecyclerView;
+import android.util.AttributeSet;
+import android.view.View;
+
+/**
+ * RecyclerView with setEmptyView method.
+ */
+public class EmptyViewRecyclerView extends RecyclerView {
+    private View emptyView;
+    final private AdapterDataObserver observer = new AdapterDataObserver() {
+        @Override
+        public void onChanged() {
+            checkIfEmpty();
+        }
+
+        @Override
+        public void onItemRangeInserted(int positionStart, int itemCount) {
+            checkIfEmpty();
+        }
+
+        @Override
+        public void onItemRangeRemoved(int positionStart, int itemCount) {
+            checkIfEmpty();
+        }
+    };
+
+    public EmptyViewRecyclerView(Context context) {
+        super(context);
+    }
+
+    public EmptyViewRecyclerView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public EmptyViewRecyclerView(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+    }
+
+    @Override
+    public void setAdapter(Adapter adapter) {
+        final RecyclerView.Adapter oldAdapter = getAdapter();
+
+        if (oldAdapter != null) {
+            oldAdapter.unregisterAdapterDataObserver(observer);
+        }
+
+        super.setAdapter(adapter);
+
+        if (adapter != null) {
+            adapter.registerAdapterDataObserver(observer);
+        }
+
+        checkIfEmpty();
+    }
+
+    void checkIfEmpty() {
```

Comment:
This method is only used within `EmptyViewRecyclerView`, so I think we can make it private.
Also, the name `checkIfEmpty` is a bit misleading, in my opinion - reading the name, I would expect boolean return value, or something similar. Maybe calling it something like `toggleEmptyView` would be better?